lib/notifications/notification-handlers.ts

```diff
@@ -0,0 +1,49 @@
+import prisma from "@/lib/prisma";
+import { Event } from "@prisma/client";
+import { sendViewedDocumentEmail } from "../emails/send-viewed-document";
+
+interface IHandleLinkViewed {
+  receiverId: string;
+  event: Event;
+  data: {
+    documentName: string;
+    viewerEmail: string | null;
+    documentId: string;
+    documentOwner: string;
+    link: {
+      enableNotification: boolean;
+    };
+  };
+}
+
+export async function handleLinkViewed(eventData: IHandleLinkViewed) {
+  try {
+    const viewerEmail = eventData.data.viewerEmail;
+    const documentName = eventData.data.documentName;
+    const message = viewerEmail
+      ? `${viewerEmail} viewed your ${documentName}`
+      : `Someone viewed your ${documentName}`;
+    const notification = await prisma.notification.create({
+      data: {
+        userId: eventData.receiverId,
+        event: eventData.event,
+        message,
+        documentId: eventData.data.documentId,
```

Comment:
The `linkId` is missing. I think it's important to record which exact link was viewed on a document.